swarm/src/dial_opts.rs

```diff
@@ -323,3 +346,18 @@ pub enum PeerCondition {
     /// configured connection limits.
     Always,
 }
+
+/// The port mode to use. I.e. on which port an outgoing connection should dial.
+/// Irrelevant when the transport doesn't support that option.
+#[derive(Default, Debug, Clone, Copy)]
+pub enum PortMode {
+    /// Leave the decision to the swarm:
+    /// - If we have 0 listen addresses, we allocate a new port by default.
+    /// - If we have at least one address, we reuse an existing port.
+    #[default]
+    Auto,
+    /// Always allocate a new port
+    New,
+    /// Always reuse an existing port
+    Reuse,
+}
```

Comment:
I think I'd prefer to not have this enum and instead store a `libp2p_core::PortMode` inside `DialOpts` as an `Option<PortMode>` where `None` means default.

Reply:
I have implemented that change